src/System/Terminal/Render.hs

```diff
@@ -27,6 +29,42 @@ type MonadCursor t m m' =
     , MonadState Cursor m
     )
 
+-- | Get the column from the 'MonadCursor' state.
+getCol :: (MonadCursor t m m') => m Int
+getCol = do
+    Cursor{col} <- get
+    pure col
+
+-- | Get the row from the 'MonadCursor' state.
+getRow :: (MonadCursor t m m') => m Int
+getRow = do
+    Cursor{row} <- get
+    pure row
+
+-- | Set the row of the 'MonadCursor' state.
+setRow :: (MonadCursor t m m') => Int -> m ()
+setRow row = do
+    Cursor{col} <- get
+    put Cursor{row = row, col = col}
+
+-- | Set the row of the 'MonadCursor' state.
+setCol :: (MonadCursor t m m') => Int -> m ()
+setCol col = do
+    Cursor{row} <- get
+    put Cursor{row = row, col = col}
+
+-- | Modify the row of the 'MonadCursor' state.
+modifyRow :: (MonadCursor t m m') => (Int -> Int) -> m ()
+modifyRow f = do
+    Cursor{row, col} <- get
+    put Cursor{row = f row, col = col}
+
+-- | Modify the column of the 'MonadCursor' state.
+modifyCol :: (MonadCursor t m m') => (Int -> Int) -> m ()
+modifyCol f = do
+    Cursor{row, col} <- get
+    put Cursor{row = row, col = f col}
```

Comment:
Aren't these already present as `gets`, `sets`, etc.?

Reply:
`gets` works, but without exported field accessors or lenses, it wouldn't be fun for users to use the function. Additionally, using `gets` here requires a deeper level of understanding of the internals on the part of the library user.

`sets` is a `lens` abstraction. there is no `puts` or `sets` in `mtl`.